ably/rest/channel.py

```diff
@@ -27,21 +28,54 @@ def __init__(self, channel):
         self.__binary = not channel.ably.options.use_text_protocol
         self.__http = channel.ably.http
 
-    def get(self):
-        path = '%s/presence' % self.__base_path
+    def _path_with_qs(self, rel_path, qs=None):
+        path = rel_path
+        if qs:
+            path += ('?' + urlencode(qs))
+        return path
+
+    def _ms_since_epoch(self, dt):
+        epoch = datetime.utcfromtimestamp(0)
+        delta = dt - epoch
+        return int(delta.total_seconds() * 1000)
+
+    def get(self, limit=None):
+        qs = {}
+        if limit:
+            qs['limit'] = min(limit, 1000)
+        path = self._path_with_qs('%s/presence' % self.__base_path.rstrip('/'), qs)
         headers = HttpUtils.default_get_headers(self.__binary)
-        response = self.__http.get(path, headers=headers)
-        return presence_response_handler(response)
+        return PaginatedResult.paginated_query(
+            self.__http,
+            path,
+            headers,
+            presence_response_handler)
 
-    def history(self):
-        url = '/presence/history'
+    def history(self, limit=None, direction=None, start=None, end=None):
+        qs = {}
+        if limit:
+            qs['limit'] = min(limit, 1000)
+        if direction:
+            qs['direction'] = direction
+        if start:
+            if isinstance(start, int):
+                qs['start'] = start
+            else:
+                qs['start'] = self._ms_since_epoch(start)
+        if end:
+            if isinstance(end, int):
+                qs['end'] = end
+            else:
+                qs['end'] = self._ms_since_epoch(end)
 
+        if 'start' in qs and 'end' in qs and qs['start']  > qs['end']:
```

Comment:
(RSC6b1) start and end are timestamp fields represented as milliseconds since epoch, or where suitable to the language, Date or Time objects. **start must be equal to or less than end** and is unaffected by the request direction.

Can you confirm this?